File: fox-hunt-domain/src/main/java/com/itechart/foxhunt/domain/entity/UserEntity.java
package com.itechart.foxhunt.domain.entity;

import com.fasterxml.jackson.annotation.JsonFormat;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonProperty.Access;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import com.fasterxml.jackson.datatype.jsr310.deser.LocalDateTimeDeserializer;
import com.fasterxml.jackson.datatype.jsr310.ser.LocalDateTimeSerializer;

import javax.validation.constraints.Email;
import java.io.Serializable;
import java.time.LocalDateTime;
import java.util.Set;
import javax.persistence.CascadeType;
import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.FetchType;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.JoinTable;
import javax.persistence.ManyToMany;
import javax.persistence.Table;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.experimental.Accessors;

@Data
@Entity
@Table(name = "app_user", schema = "fh_admin")
@NoArgsConstructor
@Accessors(chain = true)
public class UserEntity {

  @Id
  @GeneratedValue(strategy = GenerationType.IDENTITY)
  @Column(name = "app_user_id")
  private Long id;

  @Column(name = "first_name")
  private String firstName;

  @Column(name = "last_name")
  private String lastName;

  @Column(name = "date_of_birth")
  @JsonFormat(pattern = "yyyy-MM-dd'T'HH:mm:ss")
  @JsonSerialize(using = LocalDateTimeSerializer.class)
  @JsonDeserialize(using = LocalDateTimeDeserializer.class)
  private LocalDateTime dateOfBirth;

  private String country;

  private String city;

  @JsonProperty(access = Access.WRITE_ONLY)
  private String password;

  @Email
  @Column(nullable = false, unique = true)
  private String email;

  @ManyToMany(cascade = {CascadeType.MERGE}, fetch = FetchType.EAGER)
  @JoinTable(name = "organization_user_role", schema = "fh_admin",
      joinColumns = @JoinColumn(name = "user_id", referencedColumnName = "app_user_id"),
      inverseJoinColumns = @JoinColumn(name = "role_id", referencedColumnName = "role_id"))
  private Set<RoleEntity> roles;

  @Column(name = "is_activated", nullable = false)
  private boolean isActivated;

  @Column(name = "is_banned", nullable = false)
  private boolean isBanned;

  @JsonFormat(pattern = "yyyy-MM-dd'T'HH:mm:ss")
  @JsonSerialize(using = LocalDateTimeSerializer.class)
  @JsonDeserialize(using = LocalDateTimeDeserializer.class)
  @Column(name = "activated_since")
  private LocalDateTime activatedSince;

  @Column(name="avatar")
  private String avatar;

  public UserEntity(Long id) {
    this.id = id;
  }

  public UserEntity(String rootEmail) {
    this
        .setEmail(rootEmail)
        .setActivated(false);
  }
}

File: fox-hunt-api-gateway/src/common/entities/user.entity.ts
import { Type } from 'class-transformer';
import { Entity, PrimaryGeneratedColumn, Column, OneToMany } from 'typeorm';
import { OrganizationUserRoleEntity } from './organizationUserRole.entity';

@Entity('app_user', { schema: 'fh_admin' })
export class UserEntity {
  @Type(() => Number)
  @PrimaryGeneratedColumn({ name: 'app_user_id' })
  appUserId: number;

  @Column('character varying', { name: 'email', length: 40 })
  email: string;

  @Column('character varying', { name: 'password', length: 100 })
  password: string;

  @Column('boolean', { name: 'is_activated' })
  isActivated: boolean;

  @Column('character varying', { name: 'first_name' })
  firstName: string;

  @Column('character varying', { name: 'last_name' })
  lastName: string;

  @Column('character varying', { name: 'date_of_birth' })
  birthDate: string;

  @Column('character varying', { name: 'country' })
  country: string;

  @Column('character varying', { name: 'city' })
  city: string;

  @OneToMany(
    () => OrganizationUserRoleEntity,
    (organizationUserRole) => organizationUserRole.user,
  )
  organizationUserRoles: OrganizationUserRoleEntity[];
}
